internal/services/nginx/nginx_deployment_data_source.go

```diff
@@ -261,12 +291,28 @@ func (m DeploymentDataSource) Read() sdk.ResourceFunc {
 					}
 
 					if scaling := props.ScalingProperties; scaling != nil {
-						output.Capacity = pointer.ToInt64(props.ScalingProperties.Capacity)
+						if capacity := scaling.Capacity; capacity != nil {
```

Comment:
we're inconsistently using these local vars. Recommend dropping the ones that aren't used or used only once (e.g. `capacity`) and consistently using the ones we need more than onec (e.g. `scaling` instead of `props.ScalingProperties`